### ISSUE TITLE
[no-unused-expression] Throw unexpected warnings on React 16 Fragments usage.

### ISSUE THREAD

**Issue by zheeeng** [palantir/tslint#3628]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: (pick one) VSCode
-__React version__: 16.2.0

React 16 introduced Fragment api, `<></>` is a sugar usage for `<Fragment></Fragment>`, but tslint seems can't handle this syntax properly.

#### TypeScript code being linted

```ts
  render (): JSX.Element {
    return (
      < >
        <span>213</span>
        <span>213</span>
        <span>213</span>
      </>
    )
  }
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-unused-expression": true
    }
}
```

#### Actual behavior
tslint warning: `[tslint] unused expression, expected an assignment or function call (no-unused-expression)
`

#### Expected behavior
no warning.

**Comment by adidahiya** [comment]
support for fragments was added in typescript [2.6.2](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html#typescript).

**Comment by zheeeng** [comment]
@adidahiya Reinstall typescript 2.6.2, the webpack compiles pass, but still get this warning in terminal console and vscode.

**Comment by zheeeng** [comment]
@adidahiya After re-installed typescript 2.6.2 global and local, the Webpack works fine, but there are still warning in terminal and vscode.

**Comment by zheeeng** [comment]
![image](https://user-images.githubusercontent.com/1303154/34753977-10eb6bea-f5f6-11e7-86ef-e396a3438417.png)
![image](https://user-images.githubusercontent.com/1303154/34753994-29ca6882-f5f6-11e7-8072-31bf829cca39.png)
